### ISSUE TITLE
Gossfile resources are always parsed as relative

### ISSUE THREAD

**Issue by oholiab** [aelsabbahy/goss#137]
This is really just my interpretation of how I would expect the tool to work based on precedence of other tools, nothing in the codebase or documentation explicitly leads me to think that this is "wrong" as such, just counterintuitive.

Expected behaviour: gossfiles added with "goss add goss <absolute path>" will resolve as absolute paths
Actual behaviour: gossfiles are always interpreted to be relative to the gossfile being validated

I would suggest that anything with a leading '/' be interpreted as an absolute path

```
someuser@9abad7f991ab:~/golang/src/github.com/aelsabbahy/goss/release$ ls
goss-linux-386  goss-linux-amd64  shellshock_goss.yaml
someuser@9abad7f991ab:~/golang/src/github.com/aelsabbahy/goss/release$ ./goss-linux-amd64 -g /tmp/somegossfile.yaml add goss $(pwd)/shellshock_goss.yaml
Adding Gossfile to '/tmp/somegossfile.yaml':

/home/someuser/golang/src/github.com/aelsabbahy/goss/release/shellshock_goss.yaml: {}


someuser@9abad7f991ab:~/golang/src/github.com/aelsabbahy/goss/release$ ./goss-linux-amd64 -g /tmp/somegossfile.yaml validate
           
File error: open /tmp/home/someuser/golang/src/github.com/aelsabbahy/goss/release/shellshock_goss.yaml: no such file or directory
someuser@9abad7f991ab:~/golang/src/github.com/aelsabbahy/goss/release$
```

**Comment by aelsabbahy** [comment]
Fully agree with the behavior you described. I'm marking this as a bug, but I'm on vacation so will probably be a couple of weeks before I can take a look at this.

**Comment by oholiab** [comment]
No worries, I'm gonna give it a cursory look and see if I can figure it out
but failing that I have a workaround I can use :)

**Comment by aelsabbahy** [comment]
I think "file" resource handles both absolute and "~" expansion, but not 100% sure it does relative. I would have both `gossfile` and `file` use the same helper function, as the behavior should be consistent. Thanks for filing this :) and many extra thanks if you're able to PR this while I'm on vacation =)

**Comment by oholiab** [comment]
I think I've nailed it - I've just put a conditional in the json merging for now as it was a neat fix and solved my problem without having to touch either `file` or `gossfile` as it was all to do with how gossfiles were being parsed for dependencies rather than how the resource itself was treated.

I'd suggest we probably don't want this logic shared with `file` as it's probably pretty bad practice to test for a file relative to the test... Personally I'd leave it as it is :)

**Comment by oholiab** [comment]
I take back what I said about resources relative to tests - it's a pain in the ass to test your tests if they only take absolute arguments! I think I might file that as a separate issue, what do you think?

**Comment by aelsabbahy** [comment]
> I take back what I said about resources relative to tests - it's a pain in the ass to test your tests if they only take absolute arguments!

Can you give me an example of this pain point?

**Comment by oholiab** [comment]
In the case of testing your tests, you might want to reference an example file or fixtures within your own repository which will obviously change location depending on who is working on it.

I came about this problem whilst working with my fix and a sub-gossfile which had a relative path to a command which was now being parsed as relative to the entry point gossfile. I fixed this by making the path to the script absolute, but this then meant that my development instance had tests which were pointing to a script in my installed instance.

**Comment by aelsabbahy** [comment]
Interesting use case.. haven't thought about that. There's also a bit of overlap with this and #136 .

**Comment by aelsabbahy** [comment]
This issue was automatically closed when I merged the PR, feel free to re-open if you feel it's still not completely addressed.

**Comment by aelsabbahy** [comment]
Fixed in [v0.2.4](https://github.com/aelsabbahy/goss/releases/tag/v0.2.4)
